web3_proxy/src/frontend/users/payment.rs

```diff
@@ -413,60 +421,26 @@ pub async fn user_balance_post(
         // Check if the item is in the database. If it is not, then add it into the database
         let user_balance = balance::Entity::find()
             .filter(balance::Column::UserId.eq(recipient.id))
-            .one(&db_conn)
-            .await?;
+            .lock(LockType::Update)
+            .one(&txn)
+            .await?
+            .context("Could not find User balance, this should have been created when signing up the user!")?;
 
         // Get the premium user-tier
         let premium_user_tier = user_tier::Entity::find()
-            .filter(user_tier::Column::Title.eq("Premium"))
-            .one(&db_conn)
+            .filter(user_tier::Column::Title.eq(UserTier::Premium.to_string()))
```

Comment:
This will need to change to match my comments above.

Reply:
I changed it back to the previous (red) version